packages/core/upload/admin/src/components/TableList/TableRows.js

```diff
@@ -9,32 +9,44 @@ import { Typography } from '@strapi/design-system/Typography';
 import Pencil from '@strapi/icons/Pencil';
 
 import { PreviewCell } from './PreviewCell';
-import { AssetDefinition } from '../../constants';
+import { AssetDefinition, FolderDefinition } from '../../constants';
 import { formatBytes, getTrad } from '../../utils';
 
-export const TableRows = ({ assets, onEditAsset, onSelectAsset, selectedAssets }) => {
+export const TableRows = ({ onEditAsset, onEditFolder, onSelectOne, rows, selected }) => {
   const { formatDate, formatMessage } = useIntl();
 
   return (
     <Tbody>
-      {assets.map((asset) => {
-        const { alternativeText, id, name, ext, size, createdAt, updatedAt, url, mime, formats } =
-          asset;
+      {rows.map((element) => {
+        const {
+          alternativeText,
+          id,
+          name,
+          ext,
+          size,
+          createdAt,
+          updatedAt,
+          url,
+          mime,
+          formats,
+          type,
+        } = element || {};
```

Comment:
What do you think about filtering out `element`s before mapping (if that case really exists)?

Reply:
right now it exists because I didn't implement loading or empty state on the Table yet (which will come in another PR, I need to check with Maeva first)
I'm not sure I understand the filtering out proposition though 👀 is it the type-mapping you proposed below?

Reply:
I first created a merged array, filling missing keys for folders, but I removed it - since we're then passing this object to edit or move I wasn't sure it was a good idea to add unnecessary keys

Reply:
What I mean is:

```
elements.filter(Boolean).map()
```

This way you can assume `element` does always exist and can never be null ...

Reply:
makes sense thanks, 
I can do that for now, but might not need it in long term since I'll probably return other renders when no elements or loading!

Reply:
update: not needed anymore since rows will never be undefined (see Josh proposition below)